hera_cal/redcal.py

```diff
@@ -789,3 +837,335 @@ def is_redundantly_calibratable(antpos, bl_error_tol=1.0):
+                                           check_every=check_every, check_after=check_after, gain=gain)
+
+    # update omnical flags and then remove degeneracies
+    rv['g_omnical'], rv['v_omnical'] = get_gains_and_vis_from_sol(omni_sol)
+    rv['gf_omnical'] = {ant: ~np.isfinite(g) for ant, g in rv['g_omnical'].items()}
+    rv['vf_omnical'] = {bl: ~np.isfinite(v) for bl, v in rv['v_omnical'].items()}
+    make_sol_finite(omni_sol)
+    rd_sol = rc.remove_degen(omni_sol, degen_sol=rv['g_firstcal'])
+    rv['g_omnical'], rv['v_omnical'] = get_gains_and_vis_from_sol(rd_sol)
+    rv['g_omnical'] = {ant: g * ~rv['gf_omnical'][ant] + rv['gf_omnical'][ant] for ant, g in rv['g_omnical'].items()}
+
+    # compute chisqs
+    data_wgts = {bl: utils.predict_noise_variance_from_autos(bl, data, 
+                 dt=(np.median(np.ediff1d(times_by_bl[bl[:2]])) * 86400.))**-1 for bl in data.keys()}
+    rv['chisq'], nObs, rv['chisq_per_ant'], nObs_per_ant = utils.chisq(data, rv['v_omnical'], data_wgts=data_wgts, 
+                                                                       gains=rv['g_omnical'], reds=reds, 
+                                                                       split_by_antpol=(rc.pol_mode in ['1pol', '2pol']))
+    rv['chisq_per_ant'] = {ant: cs / nObs_per_ant[ant] for ant, cs in rv['chisq_per_ant'].items()}
+    if rc.pol_mode in ['1pol', '2pol']:  # in this case, chisq is split by antpol
+        for antpol in rv['chisq'].keys():
+            Ngains = len([ant for ant in rv['g_omnical'].keys() if ant[1] == antpol])
+            Nvis = len([bl for bl in rv['v_omnical'].keys() if bl[2] == join_pol(antpol, antpol)])
+            rv['chisq'][antpol] /= (nObs[antpol] - Ngains - Nvis)
+    else:
+        rv['chisq'] /= (nObs + len(rv['g_omnical']) + len(rv['v_omnical']))
+    
+    return rv
+
+
+def redcal_partial_io_iteration(hd, nInt_to_load=8, pol_mode='2pol', ex_ants=[], 
+                                solar_horizon=0.0, conv_crit=1e-10, maxiter=500, 
+                                check_every=10, check_after=50, gain=.4, verbose=False):
+    '''Perform redundant calibration (firstcal, logcal, and omnical) an entire HERAData object, loading only 
+    nInt_to_load integrations at a time and skipping and flagging times when the sun is above solar_horizon.
+    
+    Arguments:
+        hd: HERAData object, instantiated with the datafile or files to calibrate. Must be loaded using uvh5.
+        nInt_to_load: number of integrations to load and calibrate simultaneously. Lower numbers save memory, 
+            but incur a CPU overhead.
+        pol_mode: polarization mode of redundancies. Can be '1pol', '2pol', '4pol', or '4pol_minV'.
+            See recal.get_reds for more information.
+        ex_ants: list of antennas to exclude from calibration and flag. Can be either antenna numbers or
+            antenna-polarization tuples. In the former case, all pols for an antenna will be excluded.
```

Comment:
Perhaps expose other arguments used in filter_reds, just for full flexibility?

Reply:
I'd like to revisit this question as part of a bigger upgrade I've had in mind that also enables calibrating off a subset of the baselines. This is a bit more work, because we probably also want to then use those calibration solutions to come up with model visibilities for all unique baselines, even if they weren't part of the redundant calibration. This can be done by averaging, but I'd like to leave it out for now. Maybe you want to make an issue?

Reply:
Let's just expose parameters.  We can put off work on solving for unlisted baselines as a future issue.

Reply:
Ok, I've exposed them all in one big bundle as **filter_reds_kwargs